### PR TITLE
📖 update supported versions for v1.3

### PR DESCRIPTION
**Pull request by ykakarap**
**What this PR does / why we need it**:

This PR updates the supported versions matrix for v1.3.x releases.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

Part of #6615

### CODE REVIEW

File: docs/book/src/reference/versions.md [hunk 1]
```diff
@@ -63,53 +63,54 @@ These diagrams show the relationships between components in a Cluster API releas
...
-| Kubernetes v1.16  | ✓                    |                      |                     |                          |
-| Kubernetes v1.17  | ✓                    |                      |                     |                          |
-| Kubernetes v1.18  | ✓                    | ✓ (only workload)    | ✓ (only workload)   | ✓ (only workload)        |
-| Kubernetes v1.19  | ✓                    | ✓                    | ✓                   | ✓ (only workload v1.2)   |
-| Kubernetes v1.20  | ✓                    | ✓                    | ✓                   | ✓                        |
-| Kubernetes v1.21  | ✓                    | ✓                    | ✓                   | ✓                        |
-| Kubernetes v1.22  | ✓ (only workload)    | ✓                    | ✓                   | ✓                        |
-| Kubernetes v1.23* |                      | ✓                    | ✓                   | ✓                        |
-| Kubernetes v1.24  |                      |                      |                     | ✓                        |
+|                   | CAPI v0.3 (v1alpha3) | CAPI v0.4 (v1alpha4) | CAPI v1.0 (v1beta1) | CAPI v1.1+v1.2+v1.3 (v1beta1) |
```
Comment (vincepri) [inline docs/book/src/reference/versions.md:66]: A few nits:

- Drop the "CAPI" prefix to all of these columns
- Split each column into a single version, without bundling versions, even if the matrix is siilar

File: docs/book/src/reference/versions.md [hunk 2]
```diff
@@ -63,53 +63,54 @@ These diagrams show the relationships between components in a Cluster API releas
...
+| Kubernetes v1.17 + etcd/v3 | ✓                    |                      |                     |                             |
+| Kubernetes v1.18 + etcd/v3 | ✓                    | ✓ (only workload)    | ✓ (only workload)   | ✓ (only workload)           |
+| Kubernetes v1.19 + etcd/v3 | ✓                    | ✓                    | ✓                   | ✓ (only workload v1.2+v1.3) |
+| Kubernetes v1.20 + etcd/v3 | ✓                    | ✓                    | ✓                   | ✓                           |
+| Kubernetes v1.21 + etcd/v3 | ✓                    | ✓                    | ✓                   | ✓                           |
+| Kubernetes v1.22 + etcd/v3 | ✓* (only workload)   | ✓                    | ✓                   | ✓                           |
+| Kubernetes v1.23 + etcd/v3 |                      | ✓*                   | ✓*                  | ✓                           |
+| Kubernetes v1.24 + etcd/v3 |                      |                      |                     | ✓                           |
 
 The Kubeadm Control Plane Provider talks to the API server and etcd members of every Workload Cluster whose control plane it owns. It uses the etcd v3 API.
```
Comment (vincepri) [inline docs/book/src/reference/versions.md:116]: Can we add a sub-heading here that says "CoreDNS"?

File: docs/book/src/reference/versions.md [hunk 3]
```diff
@@ -63,77 +63,81 @@ These diagrams show the relationships between components in a Cluster API releas
...
+| Kubernetes v1.23 + etcd/v3 |                      | ✓*                   | ✓*                  | ✓                  | ✓                   | ✓                   |
+| Kubernetes v1.24 + etcd/v3 |                      |                      |                     | ✓                  | ✓                   | ✓                   |
 
 The Kubeadm Control Plane Provider talks to the API server and etcd members of every Workload Cluster whose control plane it owns. It uses the etcd v3 API.
 
 The Kubeadm Control Plane requires the Kubeadm Bootstrap Provider.
 
 \*  Newer versions of CoreDNS may not be compatible as an upgrade target for clusters managed with Cluster API. Kubernetes versions marked on the table are supported as an upgrade target only if CoreDNS is not upgraded to the latest version supported by the respective Kubernetes version. The versions supported are represented in the below table.
 
+#### CoreDNS
```
Comment (sbueringer) [inline docs/book/src/reference/versions.md:121]: nit: This should be a sub section of KCP. Maybe let's use: **CoreDNS**: ? So it's clearly a subsection not sure if 5 #'s is distinguishable :)

Comment (sbueringer) [inline docs/book/src/reference/versions.md:121]:
```suggestion
##### CoreDNS
```
nit: should be a sub-section of KCP

**Comment by sbueringer** [comment]
Blocked by https://github.com/kubernetes-sigs/cluster-api/pull/6872

**Comment by sbueringer** [comment]
/lgtm

fix works

**Pull request merged**